infra/src/main/kotlin/fi/fta/geoviite/infra/publication/PublicationDao.kt

```diff
@@ -310,31 +311,41 @@ class PublicationDao(
               lt.official_id,
               lt.row_id,
               lt.row_version,
-              array(select distinct v from unnest(
-                array_agg(s.switch_id) || array_agg(lt.topology_start_switch_id) || array_agg(lt.topology_end_switch_id)
-              ) as t(v) where v in (:switch_ids)) as switch_ids
+              array_agg(distinct switch_id) as switch_ids
               from (select *
-                    from layout.location_track_in_layout_context('OFFICIAL', null) lt
+                    from layout.location_track_in_layout_context('OFFICIAL', :design_id) lt
                      where not ($draftTrackIncludedCondition)
                     union all
                     select *
-                    from layout.location_track_in_layout_context('DRAFT', null) lt
+                    from layout.location_track_in_layout_context('DRAFT', :design_id) lt
                      where ($draftTrackIncludedCondition)
                     ) lt
-                left join layout.segment_version s on s.alignment_id = lt.alignment_id
-                and s.alignment_version = lt.alignment_version
+              join (
+                select location_track.id as row_id, sv_switch.switch_id
```

Comment:
Tämä on muuten rakenne joka taitaa toistua ihan kohtuullisen usein. Jos näin toteutettuna se on tehokas niin kannattaisikohan tästä tehdä näkymä (esim `location_track_switch(locationtrack_id, switch_id)`) jotta näissä sql:ssä tulee se aina samalla tavalla ja pysyy siistimpänä?